app/src/ActionModal/components/ActionModal.vue

```diff
@@ -699,6 +699,7 @@ export default {
           network: this.network,
           signingType: this.selectedSignMethod,
           password: this.password,
+          polkadotAPI: await this.$store.state.connection.polkadotAPI,
```

Comment:
use `getPolkadotApi` here

Reply:
then what is the point of store? Wasn't the idea to connect there on network switch? Actually I think the best idea was to have the `polkadotAPIs` object in there. Then you can store all of them along a session. Let's do that better

Reply:
My engineering 2 cents

Reply:
Or you mean, delete it all from store and have the function just here? (still think the `polkadotAPIs` solution in store is better)

Reply:
why maintain the api object twice? if the api object is in the utils function, that is enough, isn't it? and this way you can reuse the script in FE,API,extension.

Reply:
oh, saw this now. Yes, that is true. Taking advantage of the monorepo :thinking: 

I see you took over :+1: